### PR TITLE
adding more functionality

### PR DESCRIPTION
**Pull request by loganwright**

### CODE REVIEW

File: src/Async/Qu.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
+//
+//  Qu.swift
+//
+//  Created by Logan Wright on 10/24/15.
+//  Copyright © 2015 lowriDevs. All rights reserved.
+//
+
+public typealias Block = () -> ()
```
Comment (brightredchilli) [inline src/Async/Qu.swift:8]: is it not dangerous to do a public typealias without prefix here?

File: src/Async/Qu.swift [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+}
+func +=(operationQueue: NSOperationQueue, operation: NSOperation) {
+    operationQueue.addOperation(operation)
+}
+
+// MARK: NSOperationQueue+Operations
+
+private extension NSOperationQueue {
+    
+    /// This is the most recently added operation.  According to the docs, `operation` is returned in the order they were added to the queue, NOT the order in which they are executed.
```
Comment (brightredchilli) [inline src/Async/Qu.swift:179]: you mean 'operations' here?

File: src/Async/Qu.swift [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    }
+    
+    func Then(block: Block) -> Self {
+        return ThenAfter(.Last, block: block)
+    }
+    
+    func ThenAfter(dependency: Dependency, block: Block) -> Self {
+        let blockOp = Operation(block: block)
+        switch dependency {
+        case .Last:
```
Comment (brightredchilli) [inline src/Async/Qu.swift:94]: might be a formatting foobar here, ps...

File: src/Colors/ColorDescriptor.swift [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+    
+    // MARK: Initializers
+    
+    init(unicodeScalarLiteral value: UnicodeScalarLiteralType) {
+        self.init(value)
+    }
+    
+    init(extendedGraphemeClusterLiteral value: ExtendedGraphemeClusterLiteralType) {
```
Comment (brightredchilli) [inline src/Colors/ColorDescriptor.swift:55]: wow these are painful

File: src/StandardLibrary/String/String+Substrings.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import Foundation
+
+extension String {
+    subscript(range: Range<Int>) -> String {
+        let chars = Array(characters)
+        let substringCharacters = chars[range]
+        return String(substringCharacters)
+    }
+    
+    mutating func dropFirst() {
```
Comment (brightredchilli) [inline src/StandardLibrary/String/String+Substrings.swift:18]: guard if out of bounds, throws? :monkey:

File: src/Async/Qu.swift (hunk 3 above)
Comment (loganwright) [inline src/Async/Qu.swift:94]: This is how switch statements are auto-formatted in Swift, I generally don't fight it :)

File: src/Colors/ColorDescriptor.swift (hunk 4 above)
Comment (loganwright) [inline src/Colors/ColorDescriptor.swift:55]: Yea, it's nice once they're setting them up, but getting it going is a pain

File: src/StandardLibrary/String/String+Substrings.swift (hunk 5 above)
Comment (loganwright) [inline src/StandardLibrary/String/String+Substrings.swift:18]: Good call, I don't think throwing is as appropriate as a guard, I'll add that in!

File: src/Async/Qu.swift (hunk 2 above)
Comment (loganwright) [inline src/Async/Qu.swift:179]: Yup, fixed.

File: src/Async/Qu.swift (hunk 1 above)
Comment (loganwright) [inline src/Async/Qu.swift:8]: Nope, typealiases are also namespaced, so if there's a conflicting `Block`, this one will be `IntrepidSwiftWisdom.Block`

Comment (brightredchilli) [inline src/Async/Qu.swift:8]: Cool. So if we start including a new library that is also called block, all our existing references for that project will have to be renamed, something like that?

Comment (loganwright) [inline src/Async/Qu.swift:8]: If there's two libraries competing for the typealias, yes we'd have to replace them with a namespace if used.

The main reason I have it as public is that I'm using this to depict public argument types, so it kind of needs to be publicly exposed. Do you think I should just remove it and write all of the arguments out?

Comment (brightredchilli) [inline src/Async/Qu.swift:8]: nope. i think at this point we dont really know what the real problems are. you defiitely have a better understanding of this than me. I think we should take this as a note to ourselves, and at the first point where we see there is a problem we should get this modified.

**Pull request merged**